src/AccessibilityInsights.Extensions.GitHubAutoUpdate/AutoUpdate.cs

```diff
@@ -170,26 +170,37 @@ private UpdateResult Update()
         /// Production ctor - MUST be a default ctor or extensions will break
         /// </summary>
         public AutoUpdate() :
-            this(SetupLibrary.ReleaseChannel.Production, () => MsiUtilities.GetInstalledProductVersion(ExceptionReporter),
+            this(ProductionReleaseChannelProvider, () => MsiUtilities.GetInstalledProductVersion(ExceptionReporter),
                 new ProductionChannelInfoProvider(new GitHubWrapper(ExceptionReporter), ExceptionReporter))
         {
         }
 
         /// <summary>
         /// Unit testable ctor - allows dependency injection for testing
         /// </summary>
-        /// <param name="releaseChannel">The client's current release channel</param>
+        /// <param name="releaseChannelProvider">Provides the client's current release channel</param>
         /// <param name="installedVersionProvider">Method that provides the installed version string</param>
         /// <param name="channelInfoProvider">Method that provides a (potentially invalid) ChannelInfo</param>
-        internal AutoUpdate(ReleaseChannel releaseChannel, Func<string> installedVersionProvider, IChannelInfoProvider channelInfoProvider)
+        internal AutoUpdate(Func<ReleaseChannel> releaseChannelProvider, Func<string> installedVersionProvider, IChannelInfoProvider channelInfoProvider)
         {
-            _strongReleaseChannel = releaseChannel;
+            _strongReleaseChannel = releaseChannelProvider();
             _installedVersionProvider = installedVersionProvider;
             _channelInfoProvider = channelInfoProvider;
             ReleaseChannel = _strongReleaseChannel.ToString();
             _initTask = Task.Run(() => InitializeWithTimer());
         }
 
+        private static ReleaseChannel ProductionReleaseChannelProvider()
```

Comment:
can we consider renaming so it doesn't have Production in the name?

Reply:
Renamed to ConfiguredReleaseChannelProvider

---
In reply to: [273264454](https://github.com/Microsoft/accessibility-insights-windows/pull/277#discussion_r273264454) [](ancestors = 273264454)